# les statistiques

## 1. Introduction

**la statistique** est une science qui fait la collecte,l'interpretation et l'analyses des données pour repondre à des questions bien precises

**concepts importants :**
1. l'individu : est une unité de la population, il peux etre une personne,un produit,...
2. La population : est un ensemble de plusieurs individus de meme nature,
3. l'échantillon : est une partie representative d'une population,
4. sont des caracteristiques à une population (datatest) ex: mode,moyenne,mediane,max...

#### comment transformer le code en un outils de stat ? 

Pour se faire on a besoin d'un language de programmation pour booster  nos capacitées  et nos calculs,
ainsi que pour la visualisation. Et comme on le sait les languages des programmations populaires pour les statistiques sont le **R** et le **python** mais sans vpus mentir je ne code qu'avec le python qui est mon language de coeur ainsi li faut installer les librairies suivantes:
``` python
    pip install pandas matplotlib seaborn numpy
```
ou alors  faites ceci pour etre plus rapide : 
```python
    pip install -r requirements.txt
```



In [167]:
# importations des packages
try:
    import pandas as pd
    import numpy as np
    import matplotlib.pyplot as plt
    import seaborn as sns
except ImportError:
    print("une des librairies est introuvable ou non installer")

In [168]:
# avoir des graphiques pus lisibles et pus jolis
sns.set(style='whitegrid') 


#### chargement de la base des données


In [169]:
try:
    df = pd.read_csv("data/datatest.csv")
    
except FileNotFoundError:
    print("chemin introuvable")
df.head()

,YearsExperience,Salary
0,1.1,39343.0
1,1.3,46205.0
2,1.5,37731.0
3,2.0,43525.0
4,2.2,39891.0


In [170]:
# vu d'ensemble sur nos variables
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 2 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   YearsExperience  30 non-null     float64
 1   Salary           30 non-null     float64
dtypes: float64(2)
memory usage: 612.0 bytes


D'après ce qu'on a vu nos deux colonnes sont du type ***quantitatifs continus***
donc ça sera simple d'appliquer nos formulkes et nos hypothèse ,car ce datatest est simple ,tres pétit et orineté petits exercices   

In [171]:
# on recupere la colonne salary 

cible = df['Salary']
cible.describe()

count        30.000000
mean      76003.000000
std       27414.429785
min       37731.000000
25%       56720.750000
50%       65237.000000
75%      100544.750000
max      122391.000000
Name: Salary, dtype: float64

## 2. les stats basiques et analyses

In [172]:
moyenne = cible.mean() # moyenne
mediane = cible.median() # plus fiable
vmax = cible.max()
vmin = cible.min()
mode = cible.mode() # mode : la variable ayant pus des valeurs dans la population 

In [173]:
print(f"la moyenne des salaires est de :{moyenne}")
print(f"la mediane (valeure du milieu) est de : {mediane}")
print(f"les modes sont : \n {mode}") # peux renvoyer pusieurs valeures


la moyenne des salaires est de :76003.0
la mediane (valeure du milieu) est de : 65237.0
les modes sont : 
 0      37731.0
1      39343.0
2      39891.0
3      43525.0
4      46205.0
5      54445.0
6      55794.0
7      56642.0
8      56957.0
9      57081.0
10     57189.0
11     60150.0
12     61111.0
13     63218.0
14     64445.0
15     66029.0
16     67938.0
17     81363.0
18     83088.0
19     91738.0
20     93940.0
21     98273.0
22    101302.0
23    105582.0
24    109431.0
25    112635.0
26    113812.0
27    116969.0
28    121872.0
29    122391.0
Name: Salary, dtype: float64


**interprétation**:
- si la moyenne est proche de la mediane la distribution est relativement symétrique
- si la moyenne est nettement < mediane -> asymetrie à droite
- si la moyenne est nettement > mediane -> asymetrie à gauche 

## 3. les mesures de dispersion

In [174]:
variance = cible.var(ddof=1)
ecart_type = cible.std(ddof=1)
etendue =  cible.max() - cible.min()
cv = (ecart_type/moyenne) * 100

print(f"la variance est de {variance:.2f}")
print(f"l'eacrt type est de {ecart_type:.2f}")
print(f"le cv est de {cv:.2f}")
print(f"l'étendue est de {etendue}")

la variance est de 751550960.41
l'eacrt type est de 27414.43
le cv est de 36.07
l'étendue est de 84660.0


## 4. mesures de position

### les quartiles

##

- Q1 : (premier quartil ) : 25 pourcents en dessous
- Q2 : (mediane) : 50 pourcents de part  et d'autres
- Q3 : (troisieme quartil) : 75 pourcents au dessus

elles aident à comprendre où se concentre la majorité des nos données

In [175]:
q1 = cible.quantile(0.25)
q2 = cible.quantile(0.5)
q3 = cible.quantile(0.75)

print(f"Q1 = {q1:.2f}")
print(f"Q2 = {q2:.2f}")
print(f"Q3 = {q3:.2f}")

Q1 = 56720.75
Q2 = 65237.00
Q3 = 100544.75


#### interprétation

1. 75% des salaires sont en dessous de 56720.7 et 25%  au dessus
2. c'est la mediane 
3. 75% sont au dessus de 100544 et 25 en dessous 
 

### les deciles

les ***deciles*** sont des indicateurs statistiques qui qui decoupe la série en 10 segment
(D1,D2,D3,D4,D5,D6,...D9) 

In [ ]:
# calculs python

